File: buffer/gpu/src/jvmMain/java/com/wsr/gpu/JMatMul.java
package com.wsr.gpu;

class JMatMul {
    public native void matMul(
            long x,
            boolean transX,
            long y,
            boolean transY,
            int m,
            int n,
            int k,
            int b,
            long result,
            long runtime
    );
}

File: buffer/gpu/src/jvmAndroidMain/kotlin/com/wsr/knist/gpu/linalg/JMatMul.kt
package com.wsr.knist.gpu.linalg

object JMatMul {
    external fun matMul(
        x: Long,
        transX: Boolean,
        y: Long,
        transY: Boolean,
        m: Int,
        n: Int,
        k: Int,
        b: Int,
        result: Long,
        runtime: Long,
    )
}
